tests/sentry/workflow_engine/handlers/action/notification/test_metric_alert.py

```diff
@@ -0,0 +1,390 @@
+
+import pytest
+from django.utils import timezone
+
+from sentry.incidents.models.alert_rule import AlertRuleDetectionType, AlertRuleThresholdType
+from sentry.incidents.models.incident import IncidentStatus
+from sentry.incidents.typings.metric_detector import (
+    AlertContext,
+    MetricIssueContext,
+    NotificationContext,
+)
+from sentry.issues.grouptype import MetricIssuePOC
+from sentry.issues.issue_occurrence import IssueOccurrence
+from sentry.models.group import GroupStatus
+from sentry.models.organization import Organization
+from sentry.snuba.models import SnubaQuery
+from sentry.types.group import PriorityLevel
+from sentry.workflow_engine.handlers.action.notification.metric_alert import (
+    BaseMetricAlertHandler,
+    PagerDutyMetricAlertHandler,
+)
+from sentry.workflow_engine.models import Action
+from sentry.workflow_engine.types import WorkflowJob
+from tests.sentry.workflow_engine.test_base import BaseWorkflowTest
+
+
+class TestHandler(BaseMetricAlertHandler):
+    @classmethod
+    def send_alert(
+        cls,
+        notification_context: NotificationContext,
+        alert_context: AlertContext,
+        metric_issue_context: MetricIssueContext,
+        organization: Organization,
+        notification_uuid: str,
+    ) -> None:
+        pass
+
+
+class MetricAlertHandlerBase(BaseWorkflowTest):
+    def create_issue_occurrence(
+        self,
+        initial_issue_priority: int | None = None,
+        level: str = "error",
+        evidence_data: Mapping[str, Any] | None = None,
+    ):
+        if evidence_data is None:
+            evidence_data = {}
+
+        return IssueOccurrence(
+            id=str(uuid.uuid4()),
+            project_id=self.project.id,
+            event_id=str(uuid.uuid4()),
+            fingerprint=["test_fingerprint"],
+            issue_title="test_issue_title",
+            subtitle="test_subtitle",
+            resource_id="test_resource_id",
+            evidence_data=evidence_data,
+            evidence_display=[],
+            type=MetricIssuePOC,
+            detection_time=timezone.now(),
+            level=level,
+            culprit="test_culprit",
+            initial_issue_priority=initial_issue_priority,
+            assignee=None,
+        )
+
+    def assert_notification_context(
+        self,
+        notification_context: NotificationContext,
+        integration_id: int | None = None,
+        target_identifier: str | None = None,
+        target_display: str | None = None,
+        sentry_app_config: list[dict[str, Any]] | dict[str, Any] | None = None,
+        sentry_app_id: str | None = None,
+    ):
+        assert notification_context.integration_id == integration_id
+        assert notification_context.target_identifier == target_identifier
+        assert notification_context.target_display == target_display
+        assert notification_context.sentry_app_config == sentry_app_config
+        assert notification_context.sentry_app_id == sentry_app_id
```

Comment:
Is this everything in the notification context? Could we just do a comparison to the dict? The difference would be how test failures get bubbled up, if the integration_id and the sentry_app_id are wrong in this example, we would have to run the test again after fixing the integration_id implementation. 

That starts to add up if you have to do that many times to fix a single test.